### DISCUSSION TITLE
Trying to implement signup auth in Vue but getting 'supabaseKey is required'

### DISCUSSION THREAD

**Discussion by treyg** [supabase/supabase#1564] [Questions] [answered]
I'm trying to implement a simple signup via email in a vue component using the example in the docs, but it seems like the my key isn't being read from my .env file for some reason. Any help would be appreciated. Thanks!

```import { createClient } from "@supabase/supabase-js";
const supabaseUrl = "https://terbqcgxfxoyxgztbilu.supabase.co";
const supabaseKey = process.env.SUPABASE_KEY;
const supabase = createClient(supabaseUrl, supabaseKey);

export default {
  name: "Login",
  data() {
    return {
      email: null,
      password: null,
    };
  },
  methods: {
    registerUser() {
      const { user, session, err } = supabase.auth
        .signUp({
          email: this.email,
          password: this.password,
        })
        .then((response) => {
          console.log(user, session);
          return response;
        })
        .catch((error) => {
          if (error) {
            console.log(err);
          }
        });
    },
  },
};

**Comment by prescience-data** [comment, marked as answer]
Env files won't survive bundling (as there is no `.env` file within the browser context, that is a `node` concept). You will need to load your config with another strategy.

Hmm, maybe see these Vue docs: https://cli.vuejs.org/guide/mode-and-env.html#using-env-variables-in-client-side-code

**Reply by awalias** [reply]
yes it sounds like `process.env.SUPABASE_KEY` isn't being picked up by the app

**Reply by treyg** [reply]
That was it, thank you! I had forgotten that the Vue cli had its own method for handling .env variables.
